.github/workflows/fork-sonar.yml

```diff
@@ -0,0 +1,208 @@
+            echo "Jacoco report found"
+          else
+            echo "Warning: Jacoco report not found at expected location"
+          fi
+          echo "Finding sources and binaries..."
+          SOURCES=$(find . -type d -path "*/main/java" | sort -u | paste -sd "," -)
+          if [ -z "$SOURCES" ]; then
+            echo "Warning: No source directories found!"
+          else
+            echo "SOURCES : $SOURCES"
+            echo "SOURCES=$SOURCES" >> $GITHUB_ENV
+          fi
+          GENERATED=$(find . -type d -path "*/target/generated-sources" | sort -u | paste -sd "," -)
+          if [ -z "$GENERATED" ]; then
+            echo "Warning: No generated source directories found!"
+          else
+            echo "GENERATED : $GENERATED"
+            echo "GENERATED=$GENERATED" >> $GITHUB_ENV
+          fi
+          TESTS=$(find . -type d -path "*/test/java" | sort -u | paste -sd "," -)
+          if [ -z "$TESTS" ]; then
+            echo "Warning: No test directories found!"
+          else
+            echo "TESTS : $TESTS"
+            echo "TESTS=$TESTS" >> $GITHUB_ENV
+          fi
+          BINARIES=$(find . -type d -path "*/target/classes" | sort -u | paste -sd "," -)
+          if [ -z "$BINARIES" ]; then
+            echo "Warning: No binaries directories found!"
+          else
+            echo "BINARIES : $BINARIES"
+            echo "BINARIES=$BINARIES" >> $GITHUB_ENV
+          fi
+          LIBRARIES="distribution-core/target/dependency"
+          if [ -z "$LIBRARIES" ]; then
+            echo "Warning: No libraries directory found!"
+          else
+            echo "LIBRARIES : $LIBRARIES"
+            echo "LIBRARIES=$LIBRARIES" >> $GITHUB_ENV
+          fi
+
+      - name: Run Sonar Analysis
+        uses: SonarSource/sonarqube-scan-action@2500896589ef8f7247069a56136f8dc177c27ccf # v5.2.0
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        with:
+          args: >
+            -Dsonar.projectKey=com.powsybl:powsybl-core
+            -Dsonar.organization=powsybl-ci-github
+            -Dsonar.sources="${{ env.SOURCES }}"
+            -Dsonar.generatedSources="${{ env.GENERATED }}"
+            -Dsonar.tests="${{ env.TESTS }}"
+            -Dsonar.java.binaries="${{ env.BINARIES }}"
+            -Dsonar.java.libraries="${{ env.LIBRARIES }}"
+            -Dsonar.java.test.libraries="${{ env.LIBRARIES }}"
+            -Dsonar.coverage.jacoco.xmlReportPaths="distribution-core/target/site/jacoco-aggregate/jacoco.xml"
+            -Dsonar.pullrequest.key="${{ env.PR_NUMBER }}"
+            -Dsonar.pullrequest.branch="$(git rev-parse --abbrev-ref HEAD)"
+            -Dsonar.pullrequest.base="${{ env.BASE_REF }}"
+            -Dsonar.host.url=https://sonarcloud.io
+            -Dsonar.scm.provider=git
+
+      - name: Delete artifacts used in analysis
+        if: always()
+        uses: actions/github-script@v7
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
It would be better to use
```
with:
  github-token: ${{ secrets.GITHUB_TOKEN }}
```

(See ''Comment on PR" step in [this example](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#using-data-from-the-triggering-workflow))

Reply:
Sorry, I should have been more explicit. The `github-token` should be declared in the `with:` clause.